eclipse-theia/theia
# ui/ux: styling input validation

#### What it does

This PR fixes #12243 by providing visual feedback for input validation that is consistent with the input validation in the `search-in-workspace` and the `preferences` views. It adds this input validation for the `SingleTextInputDialog` to create better feedback for the user.

#### How to test

1. Try causing an error when renaming a file
2. Try causing an error when creating a file from the workspace using `New File`
3. Try causing an error when creating a new folder from the workspace using `New Folder`
4. Try causing an error when editing key bindings using `File` -> `Preferences` -> `Keyboard Shortcuts`
5. The error styling should be consistent with the one in `Preferences` and `search-in-workspace`

Preferences Error Validation: 
![MicrosoftTeams-image](https://github.com/eclipse-theia/theia/assets/86936229/33e38598-d539-4a11-8367-b6f8c5b6787a)

Easy errors to test include: creating a file/folder with no name or a duplicate name, renaming a file to the same name, creating empty key bindings ...


#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
